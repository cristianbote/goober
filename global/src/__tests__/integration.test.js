import { h, createContext, render } from 'preact';
import { useContext } from 'preact/hooks';
import { setup, extractCss } from 'goober';
import { createGlobalStyles } from '../index';

describe('createGlobalStyles', () => {
    it('regular', () => {
        setup(h);

        const target = document.createElement('div');

        const GlobalStyle = createGlobalStyles`
            html, body {
                background: dodgerblue;
            }
        `;

        render(
            <div>
                <GlobalStyle />
            </div>,
            target
        );

        expect(extractCss()).toMatchSnapshot();
        expect(target.innerHTML).toMatchSnapshot();
    });

    it('with theme', () => {
        const ThemeContext = createContext();
        const useTheme = () => useContext(ThemeContext);

        setup(h, null, useTheme);

        const target = document.createElement('div');

        const GlobalStyle = createGlobalStyles`
            html, body {
                margin: 0;
                background: ${(props) => props.theme.color};
            }
        `;

        render(
            <ThemeContext.Provider value={{ color: 'blue' }}>
                <div>
                    <GlobalStyle />
                </div>
            </ThemeContext.Provider>,
            target
        );

        expect(extractCss()).toMatchSnapshot();
        expect(target.innerHTML).toMatchSnapshot();
    });
});
