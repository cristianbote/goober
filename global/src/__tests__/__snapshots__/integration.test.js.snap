// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createGlobalStyles regular 1`] = `" html, body{background:dodgerblue;}"`;

exports[`createGlobalStyles regular 2`] = `"<div></div>"`;

exports[`createGlobalStyles with theme 1`] = `"html, body{margin:0;background:blue;}"`;

exports[`createGlobalStyles with theme 2`] = `"<div></div>"`;
